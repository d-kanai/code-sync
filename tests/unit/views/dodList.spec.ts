import "jest-canvas-mock";
import { mountWithFlushPromise } from "@/../tests/unit/helper";
import DoDListPage from "@/views/DoDList.vue";
import { mockDoDListApi, mockCreateDoDApi } from "@/../tests/unit/mockApi"

describe("DoDList.vue", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  it("snapshot", async () => {});

  //@IMPROVE add snapshot test
  it("should show DoD List", async () => {
    //given
    mockDoDListApi();
    //when
    const wrapper = await mountWithFlushPromise(DoDListPage);
    //then
    expect(wrapper.text()).toMatch("Long Method Mock");
    expect(wrapper.text()).toMatch("Coverage Mock");
  });
  it("should create DoD", async () => {
    //given
    mockDoDListApi();
    const _mockCreateDoDApi = mockCreateDoDApi()
    //when
    const wrapper = await mountWithFlushPromise(DoDListPage);
    await wrapper.find("#input-name").setValue("Long Method");
    await wrapper.vm.onDoDSubmit();
    //then
    expect(_mockCreateDoDApi).toHaveBeenCalledWith({ name: "Long Method" });
    expect(wrapper.text()).toMatch("Long Method");
  });
  it("validate when no input name", async () => {
    //given
    const _mockCreateDoDApi = mockCreateDoDApi()
    //when
    const wrapper = await mountWithFlushPromise(DoDListPage);
    await wrapper.vm.onDoDSubmit();
    //then
    expect(wrapper.text()).toMatch("required");
    expect(_mockCreateDoDApi).toHaveBeenCalledTimes(0);
  });
});
