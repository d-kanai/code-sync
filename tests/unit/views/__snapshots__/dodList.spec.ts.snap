// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoDList.vue snapshot 1`] = `
<div>
  <section
    class="section"
  >
    <h3
      class="subtitle"
    >
      DoDList
    </h3>
    <table
      class="table"
    >
      <thead>
        <tr>
          <th>
            ID
          </th>
          <th>
            Name
          </th>
        </tr>
      </thead>
      <tbody>
        
        <tr>
          <td>
            1
          </td>
          <td>
            <a
              class=""
              href="/dods/1"
            >
              Long Method Mock
            </a>
          </td>
        </tr>
        <tr>
          <td>
            2
          </td>
          <td>
            <a
              class=""
              href="/dods/2"
            >
              Coverage Mock
            </a>
          </td>
        </tr>
        
      </tbody>
    </table>
    <hr />
    <form>
      <div
        class="field"
      >
        <!-- &lt;label&gt;{{ label }}&lt;/label&gt; -->
        <div
          class="control"
        >
          <input
            class="input"
            id="input-name"
            placeholder="name"
          />
          <p
            class="error"
          />
        </div>
      </div>
      <button
        class="button is-primary is-small is-responsive"
        id="dodFormSubmitButton"
        type="submit"
      >
        Create DoD
      </button>
    </form>
  </section>
</div>
`;
